NoClassDefFoundError for Apache XML and EAP8

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

testsuite

### Describe the bug

Follow-up for the #https://github.com/keycloak/keycloak/pull/24633

EAP8 - OpenJDK17 | OpenJDK11 

```
11:33:24,167 INFO  [org.keycloak.testsuite.adapter.servlet.SAMLServletAdapterTest] [SAMLServletAdapterTest] salesPostEncSignedAssertionsOnlyTest() STARTED
11:33:24,569 INFO  [org.keycloak.testsuite.util.URLUtils] starting navigation
11:33:24,573 INFO  [org.keycloak.testsuite.util.URLUtils] current URL:  https://localhost:8543/auth/realms/master/super-random-page
11:33:24,573 INFO  [org.keycloak.testsuite.util.URLUtils] navigating to https://localhost:8543/auth/realms/master/super-random-page
11:33:24,576 INFO  [org.keycloak.testsuite.util.URLUtils] target URL equals current URL - refreshing the page
11:33:25,153 INFO  [org.keycloak.testsuite.util.URLUtils] new current URL:  https://localhost:8543/auth/realms/master/super-random-page
11:33:25,153 INFO  [org.keycloak.testsuite.util.URLUtils] navigation complete
11:33:25,153 INFO  [org.keycloak.testsuite.adapter.servlet.SAMLServletAdapterTest] deleting cookies in 'master' realm
11:33:25,159 INFO  [org.keycloak.testsuite.util.URLUtils] starting navigation
11:33:25,162 INFO  [org.keycloak.testsuite.util.URLUtils] current URL:  https://localhost:8543/auth/realms/master/super-random-page
11:33:25,162 INFO  [org.keycloak.testsuite.util.URLUtils] navigating to http://localhost:8280/sales-post-enc-sign-assertions-only/
11:33:25,812 INFO  [org.keycloak.testsuite.util.URLUtils] new current URL:  https://localhost:8543/auth/realms/demo/login-actions/authenticate?client_id=http%3A%2F%2Flocalhost%3A8280%2Fsales-post-enc-sign-assertions-only%2F&tab_id=2vRkrgMr9Ks
11:33:25,812 INFO  [org.keycloak.testsuite.util.URLUtils] navigation complete
&amp#27;[31m11:33:26,863 ERROR [io.undertow.request] (default task-1) UT005023: Exception handling request to /sales-post-enc-sign-assertions-only/saml: java.lang.NoClassDefFoundError: org/apache/xml/security/encryption/EncryptedData
	at org.keycloak.keycloak-saml-adapter-core-jakarta@22.0.6-SNAPSHOT-eap8-testing-22-20231116-104339//org.keycloak.adapters.saml.profile.AbstractSamlAuthenticationHandler.getAssertionFromResponse(AbstractSamlAuthenticationHandler.java:534)
	at org.keycloak.keycloak-saml-adapter-core-jakarta@22.0.6-SNAPSHOT-eap8-testing-22-20231116-104339//org.keycloak.adapters.saml.profile.AbstractSamlAuthenticationHandler.handleLoginResponse(AbstractSamlAuthenticationHandler.java:391)
	at org.keycloak.keycloak-saml-adapter-core-jakarta@22.0.6-SNAPSHOT-eap8-testing-22-20231116-104339//org.keycloak.adapters.saml.profile.AbstractSamlAuthenticationHandler.handleSamlResponse(AbstractSamlAuthenticationHandler.java:252)
	at org.keycloak.keycloak-saml-adapter-core-jakarta@22.0.6-SNAPSHOT-eap8-testing-22-20231116-104339//org.keycloak.adapters.saml.profile.AbstractSamlAuthenticationHandler.doHandle(AbstractSamlAuthenticationHandler.java:145)
	at org.keycloak.keycloak-saml-adapter-core-jakarta@22.0.6-SNAPSHOT-eap8-testing-22-20231116-104339//org.keycloak.adapters.saml.profile.webbrowsersso.WebBrowserSsoAuthenticationHandler.handle(WebBrowserSsoAuthenticationHandler.java:51)
	at org.keycloak.keycloak-saml-adapter-core-jakarta@22.0.6-SNAPSHOT-eap8-testing-22-20231116-104339//org.keycloak.adapters.saml.SamlAuthenticator.authenticate(SamlAuthenticator.java:48)
	at org.keycloak.keycloak-saml-wildfly-elytron-jakarta-adapter@22.0.6-SNAPSHOT-eap8-testing-22-20231116-104339//org.keycloak.adapters.saml.elytron.KeycloakHttpServerAuthenticationMechanism.evaluateRequest(KeycloakHttpServerAuthenticationMechanism.java:102)
	at org.wildfly.security.elytron-base@2.2.1.Final//org.wildfly.security.http.util.SetMechanismInformationMechanismFactory$1.evaluateRequest(SetMechanismInformationMechanismFactory.java:119)
	at org.wildfly.security.elytron-base@2.2.1.Final//org.wildfly.security.http.util.SocketAddressCallbackServerMechanismFactory$1.evaluateRequest(SocketAddressCallbackServerMechanismFactory.java:82)
	at org.wildfly.security.elytron-base@2.2.1.Final//org.wildfly.security.auth.server.SecurityIdentityServerMechanismFactory$1.evaluateRequest(SecurityIdentityServerMechanismFactory.java:85)
	at org.wildfly.security.elytron-base@2.2.1.Final//org.wildfly.security.http.HttpAuthenticator$AuthenticationExchange.authenticate(HttpAuthenticator.java:326)
	at org.wildfly.security.elytron-base@2.2.1.Final//org.wildfly.security.http.HttpAuthenticator$AuthenticationExchange.access$800(HttpAuthenticator.java:296)
	at org.wildfly.security.elytron-base@2.2.1.Final//org.wildfly.security.http.HttpAuthenticator.authenticate(HttpAuthenticator.java:93)
	at org.wildfly.security.elytron-web.undertow-server@4.0.0.Final//org.wildfly.elytron.web.undertow.server.SecurityContextImpl.authenticate(SecurityContextImpl.java:107)
	at org.wildfly.security.elytron-web.undertow-server-servlet@4.0.0.Final//org.wildfly.elytron.web.undertow.server.servlet.ServletSecurityContextImpl.authenticate(ServletSecurityContextImpl.java:117)
	at io.undertow.servlet@2.3.7.Final//io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:55)
	at io.undertow.core@2.3.7.Final//io.undertow.server.handlers.DisableCacheHandler.handleRequest(DisableCacheHandler.java:33)
	at io.undertow.core@2.3.7.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.core@2.3.7.Final//io.undertow.security.handlers.AuthenticationConstraintHandler.handleRequest(AuthenticationConstraintHandler.java:53)
	at io.undertow.core@2.3.7.Final//io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet@2.3.7.Final//io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
	at io.undertow.servlet@2.3.7.Final//io.undertow.servlet.handlers.security.ServletSecurityConstraintHandler.handleRequest(ServletSecurityConstraintHandler.java:60)
	at io.undertow.core@2.3.7.Final//io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at org.wildfly.security.elytron-web.undertow-server-servlet@4.0.0.Final//org.wildfly.elytron.web.undertow.server.servlet.CleanUpHandler.handleRequest(CleanUpHandler.java:38)
	at io.undertow.core@2.3.7.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.wildfly.extension.undertow@29.0.0.Final//org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
	at io.undertow.core@2.3.7.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.wildfly.extension.undertow@29.0.0.Final//org.wildfly.extension.undertow.deployment.GlobalRequestControllerHandler.handleRequest(GlobalRequestControllerHandler.java:68)
	at io.undertow.servlet@2.3.7.Final//io.undertow.servlet.handlers.SendErrorPageHandler.handleRequest(SendErrorPageHandler.java:52)
	at io.undertow.core@2.3.7.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet@2.3.7.Final//io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:276)
	at io.undertow.servlet@2.3.7.Final//io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:135)
	at io.undertow.servlet@2.3.7.Final//io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:132)
	at io.undertow.servlet@2.3.7.Final//io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
	at io.undertow.servlet@2.3.7.Final//io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at org.wildfly.extension.undertow@29.0.0.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1430)
	at org.wildfly.extension.undertow@29.0.0.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1430)
	at org.wildfly.extension.undertow@29.0.0.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1430)
	at org.wildfly.extension.undertow@29.0.0.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1430)
	at io.undertow.servlet@2.3.7.Final//io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:256)
	at io.undertow.servlet@2.3.7.Final//io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:101)
	at io.undertow.core@2.3.7.Final//io.undertow.server.Connectors.executeRootHandler(Connectors.java:393)
	at io.undertow.core@2.3.7.Final//io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:859)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at org.jboss.xnio@3.8.9.Final//org.xnio.XnioWorker$WorkerThreadFactory$1$1.run(XnioWorker.java:1282)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.ClassNotFoundException: org.apache.xml.security.encryption.EncryptedData from [Module "org.keycloak.keycloak-saml-adapter-core-jakarta" version 22.0.6-SNAPSHOT-eap8-testing-22-20231116-104339 from local module loader @558bdf1f (finder: local module finder @8576fa0 (roots: /home/jenkins/workspace/universal-test-pipeline-adapters/adapter_eap_tests-rhel8-eap8-mw_openjdk17/keycloak/testsuite/integration-arquillian/tests/base/target/containers/app-server-eap8/modules,/home/jenkins/workspace/universal-test-pipeline-adapters/adapter_eap_tests-rhel8-eap8-mw_openjdk17/keycloak/testsuite/integration-arquillian/tests/base/target/containers/app-server-eap8/modules/system/layers/base,/home/jenkins/workspace/universal-test-pipeline-adapters/adapter_eap_tests-rhel8-eap8-mw_openjdk17/keycloak/testsuite/integration-arquillian/tests/base/target/containers/app-server-eap8/modules/system/add-ons/keycloak))]
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:200)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	... 49 more
```

### Version

999

### Expected behavior

Test works

### Actual behavior

Test does not work

### How to Reproduce?

Execute `SAMLServletAdapterTest.salesPostEncSignedAssertionsOnlyTest` test on `EAP8`

### Anything else?

_No response_

--- a/distribution/galleon-feature-packs/saml-adapter-galleon-pack/src/main/resources/modules/system/add-ons/keycloak/org/keycloak/keycloak-saml-adapter-core-jakarta/main/module.xml
+++ b/distribution/galleon-feature-packs/saml-adapter-galleon-pack/src/main/resources/modules/system/add-ons/keycloak/org/keycloak/keycloak-saml-adapter-core-jakarta/main/module.xml
@@ -38,6 +38,7 @@
         <module name="org.keycloak.keycloak-core"/>
         <module name="org.keycloak.keycloak-crypto-default" services="import"/>
         <module name="org.apache.httpcomponents"/>
+        <module name="org.apache.santuario.xmlsec"/>
     </dependencies>
 
 </module>

